src/controller/auth.js

```diff
@@ -10,7 +10,12 @@ const unauthError = require("../responses/unauthenticated.json");
 module.exports = router
 	.post("/register", (req, res) => {
 		registerAccount(req.body.username, req.body.password).then(response => {
-			res.json(response);
+			if (!response.ok) {
+				res.json(response);
```

Comment:
Don't forget to `return` as well, to get out of the function. Otherwise the next `res.json()` will just overwrite.